pkg/aws/apis/validation/validation_test.go

```diff
@@ -1334,6 +1334,99 @@ var _ = Describe("Validation", func() {
 					},
 				},
 			}),
+			Entry("AWS machine class with instanceMetadata", &data{
+				setup: setup{},
+				action: action{
+					spec: &awsapi.AWSProviderSpec{
```

Comment:
It will make the tests more readable if you create a function which will create and initialize `AWSProviderSpec` and then in individual tests you can set `InstanceMetadataOptions` with different values.

Reply:
The whole file is basically written like that and I wanted to avoid having to refactor all the tests, but I can do that

Reply:
I changed the last tests I added as suggested